src/fidesops/ops/service/saas_request/override_implementations/firebase_auth_request_overrides.py

```diff
@@ -0,0 +1,166 @@
+import logging
+from typing import Any, Dict, List
+
+import firebase_admin
+from firebase_admin import App, auth, credentials
+from firebase_admin.auth import UserRecord
+
+from fidesops.ops.graph.traversal import TraversalNode
+from fidesops.ops.models.policy import Policy
+from fidesops.ops.models.privacy_request import PrivacyRequest
+from fidesops.ops.service.saas_request.saas_request_override_factory import (
+    SaaSRequestType,
+    register,
+)
+from fidesops.ops.util.collection_util import Row
+
+logger = logging.getLogger(__name__)
+
+
+@register("firebase_auth_user_access", [SaaSRequestType.READ])
+def firebase_auth_user_access(
+    node: TraversalNode,
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    input_data: Dict[str, List[Any]],
+    secrets: Dict[str, Any],
+) -> List[Row]:
+    """
+    SaaS Request Override function used to integrate access requests with
+    Firebase Auth service. Specifically, it reads `User` records.
+    """
+    app = initialize_firebase(secrets)
+
+    processed_data = []
+
+    emails = input_data.get("email", [])
+    for email in emails:
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        row = {"email": user.email, "uid": user.uid}
+        keys = [
+            "email",
+            "uid",
+            "display_name",
+            "phone_number",
+            "photo_url",
+            "disabled",
+            "email_verified",
+        ]
+        for key in keys:
+            row[key] = getattr(user, key, None)
```

Comment:
this is going to add `None` values in for fields that are not populated in the response record dictionary/object. is that OK? i _think_ the behavior in "normal" saas requests is just to not include the field in the response record, in which case i think i'll need to handle this slightly differently...

@galvana @earmenda

Reply:
i'd say use `hasattr` to avoid this